src/services/balancer/subgraph/entities/pools/index.ts

```diff
@@ -79,9 +88,40 @@ export default class Pools {
     } catch {
       // eslint-disable-previous-line no-empty
     }
+
+    if (this.excludedAddresses == null) {
+      // fetch only once
+      this.excludedAddresses = await this.getExcludedAddresses();
+    }
+
     return this.serialize(pools, pastPools, period, prices, currency);
   }
 
+  public removeExcludedAddressesFromTotalLiquidity(
+    pool: Pool,
+    rawTotalLiquidity: string
+  ) {
+    let totalLiquidity = bnum(rawTotalLiquidity);
+
+    if (
+      this.excludedAddresses != null &&
+      this.excludedAddresses[pool.address] != null
+    ) {
+      Object.values(this.excludedAddresses[pool.address]).forEach(
+        accountBalance => {
+          const accountBalanceFormatted = formatUnits(accountBalance, 18);
+          const poolShare = bnum(accountBalanceFormatted).div(pool.totalShares);
+
+          totalLiquidity = totalLiquidity.minus(
+            totalLiquidity.times(poolShare)
```

Comment:
We have to multiply the share by the original `totalLiquidity`, not the value we're updating inside the loop. See 62da75b678d8f1905960925333f5404af668b1e7 for a proposed fix